authors/apps/articles/views.py

```diff
@@ -99,3 +99,140 @@ def retrieve(self, request, slug):
         serializer = self.detailSerializer()
         serializer.instance = article_instance
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+class RetrieveArticleWithReadingTime(generics.RetrieveAPIView):
+
+    def get_queryset(self):
+
+        queryset = Article.objects.all()
+
+        return queryset
+
+    def retrieve(self, *args):
+
+        articles = self.get_queryset()
+
+        article_body = []
+
+        title = {}
+
+        description = {}
+
+        body = {}
+
+        time = {}
+
+        created_at = {}
+
+        WPM = 200
+
+        for article in articles:
```

Comment:
Putting this sort of logic here works fine but violates the `Single Responsibility Principe`.

Its hard to reuse logic in views and as such, views are supposed to be delegators, most of the times, they refer to classes/services like permissions, serializers, and renderers to accomplish their tasks. They rarely do something on their own, that being said, this sort of logic would be better extracted to a serializer.